internal/atls/atls.go

```diff
@@ -248,7 +250,13 @@ func verifyEmbeddedReport(validators []Validator, cert *x509.Certificate, peerPu
 				return nil
 			}
 			// Otherwise, we'll keep track of the error and continue with the next validator.
-			retErr = errors.Join(retErr, fmt.Errorf("validator %s failed: %w", validator.OID(), validationErr))
+			// The joined error should reveal the atls nonce once to maintain readability. Because the error is only revealed if all validators fail,
+			// we can implicitly include the nonce in the first validator error and the concatenate the other errors.
```

Comment:
I think it would make more sense to first collect the errors with `Join`, and then wrap them with a message that contains the nonce before returning from this function.